src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
+ * backwards continuous
  *
- * @pre constant_period_trajectory is pre-allocated up the the limit specified by
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @param segment_lengths [in] The length of each segment in the trajectory
  *
- * @param constant_period_trajectory [out] The array to be filled with the constant time
- * interpolation period equivalent of the input trajectory
- * @param variable_time_trajectory [in] The valid trajectory used as
- * reference for a constant interpolation period trajectory
- * @param interpolation_period [in] The constant change in time [s] that corresponds to
- * each trajectory segment.
+ * @param max_allowable_acceleration [in] The maximum acceleration that can occur anywhere
+ * on the trajectory
  *
+ * @param initial_speed [in] The initial speed at the beggining of the trajectory
+ *
+ * @param num_segments [in] The number of nodes(elements) in the trajectory
+ *
+ * @return A status indicating whether or not generation was successful
  */
 TrajectoryPlannerGenerationStatus_t
-app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
-    PositionTrajectory_t* constant_period_trajectory,
-    PositionTrajectory_t* variable_time_trajectory, const float interpolation_period);
+app_trajectory_planner_modifySpeedsToBackwardsContinuous(
+    float speeds[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float max_allowable_acceleration, float initial_speed, unsigned int num_segments);
 
 /**
- * Generates the X/Y points for each position on the constant arc length trajectory
+ * Function creates a forwards continuous speed profile based on the specified parameters.
+ * The profile will never exceed the maximums speeds specified by the
+ * max_allowable_speed_profile
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
  *
- * @pre traj_elements [out] is pre-allocated up the the limit specified by
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS. Will be modified to contain the X/Y points of the
- * trajectory
+ * @param speeds [out] The pre-allocated array that will be modified to contain a forwards
+ * continuous speed profile that obeys the initial and final speeds and the
+ * max_allowable_speed_profile that limits the speed at each point on the trajectory. In
+ * m/s.
  *
- * @param path [in] The polynomial that defines the path of the trajectory
+ * @param segment_lengths [in] The length of each segment between data points on the speed
+ * profile. In meters.
  *
- * @param num_elements [in] The number of elements in the trajectory
+ * @param max_allowable_speed_profile [in] An array that limits the max speed at each
+ * point on the profile. In m/s.
  *
- * @param arc_length_parameterization [in] The arc length parameterization of the path
- * polynomial
+ * @param max_allowable_acceleration [in] The maximum allowable acceleration along any
+ * segment in the profile In m/s^2.
```

Comment:
Since this function can be used for linear or angular now, are these units relevant?

Reply:
Added comment stating how the units don't matter as long as the magnitudes remain consistant.